### ISSUE TITLE
Add TryGetFileByServerRelativeUrl methods that can handle a non existing file + remove the Exists property

### ISSUE THREAD

**Issue by jansenbe** [pnp/pnpcore#515]

**Comment by DaleyKD** [comment]
Are you going to get rid of IFile.Exists?  We use that very heavily.

**Comment by jansenbe** [comment]
Does Exists ever get set to false when you load a file that does not exist today? Someone reported that it doesn't as an exception is thrown which is not caught...hence it would never be set to false. I'll do some additional testing, but for now will not remove Exists, thanks for the quick feedback

**Comment by DaleyKD** [comment]
Honestly not sure.  I can try to give it a test if I find some time.

A lot of code has been migration from CSOM to REST, so the `.Exists` is definitely because of old CSOM.

**Comment by jansenbe** [comment]
Did some more testing and the Exists property in REST does not work as expected, trying to load a non existing file simply returns in an exception being thrown by the server. @DaleyKD : I would recommend reviewing your code paths that use Exists as they might not work as expected. 

I'll start working on a TryGetFileByServerRelativeUrl that returns false if the file does not exist and true + the IFile in case the file exists.

**Comment by DaleyKD** [comment]
@jansenbe .  Thanks!

Would returning `null` if something goes awry be better?  (Doesn't matter to me.  Just curious.)

It'll be a while before we implement this function.  I plan on "upgrading" to GA 1.3 and camping out there for a while.  This is the last week of our Programming Increment, and I'm not sure what we'll be doing next.

**Comment by jansenbe** [comment]
Let me see if I can mimic the CSOM exists behavior instead of introducing a new method.

**Comment by jansenbe** [comment]
I added a Added IWeb.GetFileByServerRelativeUrlOrDefault methods that can return null for non existing files. Did not go for the tryget... approach as the .NET convention requires it to return a bool with an out param and using out params in async methods is a no go. The "OrDefault" implementation is aligned with LINQ/EF.

@DaleyKD : as there was no way to get Exists to work reliable (especially in batch scenarios) I had to drop it. So before moving to this version you would have to review your code base.
